pkg/application/registryConfigUseCase.go

```diff
@@ -153,12 +167,30 @@ func (useCase *EcosystemRegistryUseCase) applySensitiveDoguConfigDiffs(ctx conte
 		}
 	}
 
-	errs = append(errs, callIfNotEmpty(ctx, entriesToSet, useCase.doguSensitiveConfigRepository.SaveAll))
+	errs = append(errs, callIfNotEmpty(ctx, entriesToEncrypt, useCase.doguSensitiveConfigRepository.SaveAll))
+	errs = append(errs, callIfNotEmpty(ctx, encryptedEntriesToSet, useCase.doguSensitiveConfigRepository.SaveAllForNotInstalledDogus))
 	errs = append(errs, callIfNotEmpty(ctx, keysToDelete, useCase.doguSensitiveConfigRepository.DeleteAllByKeys))
 
 	return errors.Join(errs...)
 }
 
+// Only encrypt diffs with action domain.ConfigActionSetEncrypted. Diffs with action domain.ConfigActionSetToEncrypt will
+// be encrypted by other components in further procedure.
+func (useCase *EcosystemRegistryUseCase) encryptSensitiveDoguDiffs(ctx context.Context, diffs domain.SensitiveDoguConfigDiffs) (map[common.SensitiveDoguConfigKey]common.EncryptedDoguConfigValue, error) {
+	toEncryptEntries := map[common.SensitiveDoguConfigKey]common.SensitiveDoguConfigValue{}
+	for _, diff := range diffs {
+		if diff.NeededAction == domain.ConfigActionSetEncrypted {
+			toEncryptEntries[diff.Key] = common.SensitiveDoguConfigValue(diff.Expected.Value)
+		}
+	}
```

Comment:
with a GroupByAction in the domain you can also remove this condition in the loop.